### ISSUE TITLE
[BokehJS] Missing properties active_scroll and active_drag in figure

### ISSUE THREAD

**Issue by Astrum42** [bokeh/bokeh#7109]
Hi,
using BokehJS v0.12.10, I cannot configure my plot to have wheel_zoom and pan mode enabled by default.
Would this be something easy to add to the API?
Thanks in advance!
Cheers,
A.
```
<html>
<head>
	<link rel="stylesheet" href="https://cdn.pydata.org/bokeh/release/bokeh-0.12.10.min.css" type="text/css" />
	<script type="text/javascript" src="https://cdn.pydata.org/bokeh/release/bokeh-0.12.10.min.js"></script>
	<script type="text/javascript" src="https://cdn.pydata.org/bokeh/release/bokeh-api-0.12.10.min.js"></script>
	<!-- The order of CSS and JS imports above is important. -->
	<script type="text/javascript">
	window.onload=function(){
		// data to plot
		var source = new Bokeh.ColumnDataSource({
			data: { x: [0,1,2,3,4,5,6,7,8,9], y: [0,1,4,-2,2,5,0,2,1,1] }
		});

		// make the plot and add some tools
		var tools = "pan,crosshair,wheel_zoom,box_zoom,reset,save";

		var plot = Bokeh.Plotting.figure({
			title:"demo plot",
			tools: tools,
			active_scroll:"wheel_zoom",
			active_drag:"pan",
			toolbar_location:"above",
			sizing_mode:"stretch_both"
		});
		
		var scatterData = plot.line({ field: "x" }, { field: "y" }, {
			source: source,
			line_width: 2
		});

		// Show the plot, appending it to the end of the current
		// section of the document we are in.
		Bokeh.Plotting.show(plot,document.getElementById("myPlot"));
		
	}
	</script>
</head>
<body>
	<div id="myPlot"></div>
</body>
</html>
```
Uncaught Error: property Plot.active_scroll wasn't declared
    at r.HasProps.t.setv (bokeh-0.12.10.min.js:sourcemap:1)
    at r.t (bokeh-0.12.10.min.js:sourcemap:1)
    at r.e [as constructor] (bokeh-0.12.10.min.js:sourcemap:22)
    at r.e [as constructor] (bokeh-0.12.10.min.js:sourcemap:29)
    at r.e [as constructor] (bokeh-0.12.10.min.js:sourcemap:30)
    at new r (bokeh-api-0.12.10.min.js:sourcemap:2)
    at Object.t.figure (bokeh-api-0.12.10.min.js:sourcemap:3)
    at window.onload (demo.html:17)

**Comment by mattpap** [comment]
> Would this be something easy to add to the API?

Yes. bokehjs' plotting API is close to 1-1 translation of bokeh's plotting API, so implementation of `bokeh.plotting.helpers._process_active_tools()` and respective properties on `Figure` is sufficient.

**Comment by mattpap** [comment]
On a side note, when developing with bokeh (and in general for the web) and submitting bug reports, it's beneficial to use unminified bokeh*.js files. You will get more readable tracebacks this way. Also `window.onload=function(){` is an unnecessary burden. You can avoid it like this:
```html
<html>
<head>
  <link rel="stylesheet" href="https://cdn.pydata.org/bokeh/release/bokeh-0.12.10.css" type="text/css" />
  <script type="text/javascript" src="https://cdn.pydata.org/bokeh/release/bokeh-0.12.10.js"></script>
  <script type="text/javascript" src="https://cdn.pydata.org/bokeh/release/bokeh-api-0.12.10.js"></script>
</head>
<body>
  <div id="myPlot"></div>
  <script>
    // data to plot
    var source = new Bokeh.ColumnDataSource({
      data: { x: [0,1,2,3,4,5,6,7,8,9], y: [0,1,4,-2,2,5,0,2,1,1] }
    });

    // make the plot and add some tools
    var tools = "pan,crosshair,wheel_zoom,box_zoom,reset,save";

    var plot = Bokeh.Plotting.figure({
      title:"demo plot",
      tools: tools,
      //active_scroll:"wheel_zoom",
      //active_drag:"pan",
      toolbar_location:"above",
      sizing_mode:"stretch_both"
    });

    var scatterData = plot.line({ field: "x" }, { field: "y" }, {
      source: source,
      line_width: 2
    });

    // Show the plot, appending it to the end of the current
    // section of the document we are in.
    Bokeh.Plotting.show(plot,document.getElementById("myPlot"));
  </script>
</body>
</html>
```

**Comment by Astrum42** [comment]
thanks for the hints, I'm taking note for next time.
Would you mind having a quick look at this stackoverflow question (now following your guidelines), as I am looking for an other BokehJS API function which I am not sure exists (and which I could therefore turn into an improvement proposal to be discussed in this github).
https://stackoverflow.com/questions/46956603/how-to-trigger-bokehjs-plot-resize-upon-jquery-ui-resize-event
Thanks in advance.
Cheers,
A.

**Comment by PBrockmann** [comment]
active_scroll = "auto"
gives
Error: property Plot.active_scroll wasn't declared

Tested from Observable notebook: https://observablehq.com/@pbrockmann/test-bokehjs
with 1.0.4 release

**Comment by PBrockmann** [comment]
Tested with last 1.3.4, the problem remains.
Any news on this small missing feature ?

**Comment by bryevdv** [comment]
@PBrockmann were there any news, it would be here in the issue. I don't really work on the BokehJS API, so it will not get done until @mattpap or someone else decides spend time on it.

**Comment by PBrockmann** [comment]
Yes you are right. I just wanted to mention my interest in this issue. Sorry.

**Comment by bryevdv** [comment]
@PBrockmann no apology needed, just letting you know where things stand at present. FWIW this is potentially something you could look at if you are interested, happy to try and provide guidance or info

**Comment by PBrockmann** [comment]
Just a comment to precise a test code: https://jsfiddle.net/PBrockmann/u6jnjcab/

**Comment by mattpap** [comment]
This was fixed in PR #13267 in bokeh 3.2.2.
